src/AppContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ReactNode, useState } from 'react';
+import { User } from './types/User';
+
+interface AppContextType {
+  user: User | 0,
+  setUser: (userData: User | 0) => void,
```

Comment:
Actually, it's better to use null for these purposes. What do you think about it? 
```suggestion
  user: User | null,
  setUser: (userData: User | null) => void,
```